app/src/main/java/org/fossasia/pslab/experimentsetup/electronicexperiments/DiodeExperiment.java

```diff
@@ -34,13 +34,16 @@ public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
             @Override
             public void onClick(View v) {
                 Intent intent = new Intent(getActivity(), OscilloscopeActivity.class);
-                if ("Half Wave Rectifier".equals(experiment))
-                    intent.putExtra("who", "Half Wave Rectifier");
-                else if ("Full Wave Rectifier".equals(experiment))
-                    intent.putExtra("who", "Full Wave Rectifier");
-                else
-                    intent.putExtra("who", "Diode Clipping Clamping");
-                startActivity(intent);
+                switch (experiment){
+                    case "Half Wave Rectifier" :
+                        intent.putExtra("who", "Half Wave Rectifier");
+                        break;
+                    case "Full Wave Rectifier" :
+                        intent.putExtra("who", "Full Wave Rectifier");
+                        break;
+                    default:intent.putExtra("who", "Diode Clipping Clamping");
+                             startActivity(intent);
```

Comment:
Looks good! Try to give a `Reformat Code` so all the missing spaces and indentation will be all good. :)

Reply:
Sure :+1: